rootfs/usr/share/inputplumber/schema/capability_map_v2.json

```diff
@@ -0,0 +1,1257 @@
+          "type": "string",
+          "enum": [
+            "SND"
+          ]
+        },
+        {
+          "description": "There are no events of this type, to my knowledge, but represents metadata about key repeat configuration.",
+          "type": "string",
+          "enum": [
+            "REP"
+          ]
+        },
+        {
+          "description": "Looking at the source of [`fftest`](https://github.com/flosse/linuxconsole/blob/master/utils/fftest.c) This seems to be sent to the device with a previusly obtained effect id as a code in order to toggle the effect.",
+          "type": "string",
+          "enum": [
+            "FF"
+          ]
+        },
+        {
+          "description": "I think this is unused?",
+          "type": "string",
+          "enum": [
+            "PWR"
+          ]
+        },
+        {
+          "description": "A force feedback effect's state changed.",
+          "type": "string",
+          "enum": [
+            "FF_STATUS"
+          ]
+        },
+        {
+          "description": "An event originating from uinput.",
+          "type": "string",
+          "enum": [
+            "UINPUT"
+          ]
+        }
+      ]
+    },
+    "GamepadCapability": {
+      "type": "object",
+      "properties": {
+        "axis": {
+          "anyOf": [
+            {
+              "$ref": "#/definitions/AxisCapability"
+            },
+            {
+              "type": "null"
+            }
+          ]
+        },
+        "button": {
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "gyro": {
+          "anyOf": [
+            {
+              "$ref": "#/definitions/GyroCapability"
+            },
+            {
+              "type": "null"
+            }
+          ]
+        },
+        "trigger": {
+          "anyOf": [
+            {
+              "$ref": "#/definitions/TriggerCapability"
+            },
+            {
+              "type": "null"
+            }
+          ]
+        }
+      }
+    },
+    "GyroCapability": {
+      "type": "object",
+      "required": [
+        "name"
+      ],
+      "properties": {
+        "axis": {
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "deadzone": {
+          "type": [
+            "number",
+            "null"
+          ],
+          "format": "double"
+        },
+        "direction": {
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "name": {
+          "type": "string"
+        }
+      }
+    },
+    "HidrawConfig": {
+      "description": "A [HidrawConfig] defines how to decode a particular event in an HID input report.",
+      "type": "object",
+      "required": [
+        "bit_offset",
+        "byte_start",
+        "input_type",
+        "report_id"
+      ],
+      "properties": {
+        "bit_offset": {
+          "type": "integer",
+          "format": "uint8",
```

Comment:
A byte can also be signed. Are we accounting for that?

Reply:
The `input_type` property is what will define how to decode a particular part of the input report.